Source/WebDriver/WebSocketServer.cpp

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (C) 2024 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WebSocketServer.h"
+
+#include "Session.h"
+#include "WebDriverService.h"
+#include <algorithm>
+#include <optional>
+#include <wtf/JSONValues.h>
+#include <wtf/UUID.h>
+#include <wtf/text/MakeString.h>
+#include <wtf/text/WTFString.h>
+
+namespace WebDriver {
+
+WebSocketServer::WebSocketServer(WebSocketMessageHandler& messageHandler, WebDriverService& service)
+    : m_messageHandler(messageHandler), m_service(service)
+{
+}
+
+void WebSocketServer::addToConnectionsNotAssociatedToASession(WebSocketMessageHandler::Connection connection)
+{
+    m_connectionsNotAssociatedToASession.push_back(connection);
+}
+
+void WebSocketServer::addToConnectionsAssociatedToASession(std::pair<WebSocketMessageHandler::Connection, String> connectionAndSession)
+{
+    m_connectionToSession.push_back(connectionAndSession);
+}
+
+bool WebSocketServer::isConnectionNotAssociatedToSession(WebSocketMessageHandler::Connection& connection)
+{
+    return std::find(m_connectionsNotAssociatedToASession.begin(), m_connectionsNotAssociatedToASession.end(), connection) != m_connectionsNotAssociatedToASession.end();
```

Comment:
I think this would read better with std::count or std::count_if since we don't use the returned value.

Reply:
Replaced with `std::count`.